pkg/controller/admissionchecks/multikueue/multikueuecluster.go

```diff
@@ -36,39 +37,61 @@ import (
 	"k8s.io/client-go/tools/clientcmd"
 	"k8s.io/client-go/util/workqueue"
 	"k8s.io/klog/v2"
+	"k8s.io/utils/ptr"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/event"
 	"sigs.k8s.io/controller-runtime/pkg/handler"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
 
 	kueuealpha "sigs.k8s.io/kueue/apis/kueue/v1alpha1"
 	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
 )
 
+const (
+	eventChBufferSize = 10
+
+	// this set will provide waiting time between 5s to 5min
+	retryIncrement = 5 * time.Second
+	retryMaxSteps  = 6
+)
+
+// retryAfter returns an exponentially increasing interval between
+// retryIncrement and 2^retryMaxSteps * retryIncrement
+func retryAfter(failedAttempts uint) time.Duration {
```

Comment:
> Not being able to connect should not be seen as a reconcile error in my opinion, as it is not related to k8s state.

In most cases when Kueue sends a request from node to kube API server, and the API server drops the request we handle the failure as reconcile error. 

However, this "internal" (within cluster) connect error, maybe for external connect errors longer baseDelay is preferred indeed.

> Also with this we maintain the control over the retry timing.

I see, I just have a preference for the KISS principle, we could introduce our timing mechanism later when proven to be needed. 

However, on the fence here, because maybe communication with external cluster higher `baseDelay` is preferred indeed. WDYT @alculquicondor ?

If case we want to control the timings, is it much of complication to use the standard rate limiting queue, like for example [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/job/job_controller.go#L163C3-L163C14). Then we could pass the `baseDelay` and `maxDelay`. However, if this is a big complication, I'm fine as is.

Reply:
You could also use the Backoff class from k8s.io/apimachinery/pkg/util/wait

But on the nit side.

Reply:
@trasc if you prefer to keep the custom timings, I'm fine just do a quick review if we can simplify the code by using the rate limiter or the package suggested by Aldo, so that we avoid reinventing the wheel. If you find this is the simplest approach. I'm ok, but please review the options.

Reply:
I did look at `Backoff` in k8s.io/apimachinery/pkg/util/wait but it's a bit overkill for what we are doing here.
Another thing I was thinking of was to just double the time since the cluster was declared inactive , so if it failed 5min ago , we try now and failed again retry in 5 min, the plus side of this being that we don't need to keep an internal state but the behavior is harder to predict.